### PR TITLE
[CDSK-717] Fix status label and add status sorting

### PR DESCRIPTION
**Pull request by Mowinski**
I've fixed status label. Also, I've added a sorting of dependencies by statuses.

test

### CODE REVIEW

File: www/script/test/test_helpers.js [hunk 1]
```diff
@@ -88,4 +88,50 @@ define(["jquery", "helpers"], function ($, helpers) {
...
+
+    });
+    it("sort result passed as object by 'results' key", function() {
+      var ret = helpers.sortByStatus(this.input_object_by_result, 'results', this.opts_stub);
+
+      var expected_value = '4\n1\n3\n2\n';
+      expect(ret).toBe(expected_value);
+    });
+
+    it("sort result passed as object by 'order' key", function() {
```
Comment (warcholprzemo) [inline www/script/test/test_helpers.js:152]: Duplicated test

**Review by warcholprzemo — CHANGES_REQUESTED** [review]

**Review by warcholprzemo — APPROVED** [review]

**Comment by campos-ddc** [comment]
Make sure to use uppercase JIRA project name on the next branch (`CDSK-717-fix-bug`)

Also, `fix-bug` is not a very useful branch name :P

**Review by campos-ddc — APPROVED** [review]

**Comment by campos-ddc** [comment]
https://unity.slack.com/archives/G9AD6RWES/p1520414765000005

I can no longer see the dependency names next to their status

I noticed that on your screenshots, but I thought that was just something with your local Katana instance

**Comment by campos-ddc** [comment]
https://unity.slack.com/archives/G9AD6RWES/p1520414961000371

All builds that show as "Waiting for dependency" are actually waiting for an agent for themselves (the build page says "Build will resume", which is closer to the real behavior). That status should probably be "Waiting for available slave" instead of "Waiting for dependency"

**Comment by campos-ddc** [comment]
https://unity.slack.com/archives/G9AD6RWES/p1520415038000153

This failed build doesn't show any step information https://staging-katana.bf.unity3d.com/projects/Unity/builders/proj0-Test%20GraphicsTests%20-%20Linux32Standalone%20OpenGL%20%28Core%29/builds/445?unity_branch=trunk

**Comment by campos-ddc** [comment]
https://unity.slack.com/archives/G9AD6RWES/p1520415209000467

Some builds do not show the dependency information.

I have a feeling that is because the dependencies were merged against other running builds

For cases like this, you should look for the dependency `buildrequestid`, but if that buildrequest has a `mergebrid` we should show the information for the request we merged into

**Comment by campos-ddc** [comment]
https://unity.slack.com/archives/G9AD6RWES/p1520415430000143

Builds that are *actually* waiting for dependency are showing up as RUNNING, not sure how we can actually detect if it is waiting for dependencies, perhaps look for running status and check if the current step is a Trigger step?

**Comment by Mowinski** [comment]
I've updated screenshot

File: www/prod/css/default.css [hunk 2]
```diff
@@ -728,13 +728,13 @@ td.visible-print,th.visible-print{display:table-cell!important}}
...
 .table-katana .paused{border-top:1px solid #fff}
 .log-list-paused>a:before{background-color:#DAE6F3}
-.running{background-color:#FCFACB}
-.color-blind-mode .running{background-color:#c2ba0a!important}
-.color-blind-mode .dependency-list .running:before{content:"Running"}
-.color-blind-mode .log-list-running>a:before{background-color:#c2ba0a}
-.state-text.running:before{content:"Running"}
-.table-katana .running{border-top:1px solid #fff}
-.log-list-running>a:before{background-color:#FCFACB}
+.waiting-for-dependency{background-color:#FCFACB}
```
Comment (campos-ddc) [inline www/prod/css/default.css:731]: Should this be renamed to `waiting-for-available-slave`?

**Pull request merged**